modules/ROOT/pages/reliability-patterns.adoc

```diff
@@ -0,0 +1,162 @@
+|Endpoint |Advantage |Disadvantage
+|VM (standalone Mule)
+|
+* Faster than JMS because everything is in memory and in-process.
+* Provided ready to use with Mule. There is no need to purchase additional software.
+|
+* Data is lost when Mule exits.
+* Non-clustered.
+|File-based VM (standalone Mule)
+|
+* Provided ready to use with Mule. There is no need to purchase additional software.
+|
+* Non-clustered.
+|VM (Mule HA cluster)
+|
+* Faster than JMS because everything is in shared memory.
+* Provided ready to use with Mule. There is no need to purchase additional software.
+|
+* Data is lost when the entire Mule cluster exits.
+|JDBC
+|
+* Depending on resources available, can hold a very large amount of information.
+|
+* Can be complex to use since proper DBMS schemas are required
+|JMS (not enabled for persistence)
+|
+* Allows sharing data with other Mule instances or non-Mule clients.
+|
+* It is slower than VM because it is in another process.
+|JMS (enabled for persistence)
+|
+* It is the most reliable choice because messages are persisted to disk.
+|
+* It is slower than VM because of disk access.
+|===
+
+== Implementing a Reliability Pattern
+
+Here is a code example that shows what a complete reliability pattern would look like, including the reliable acquisition flow and the application logic flow. In this example, the reliable acquisition flow part of the reliability pattern is HTTP-to-VM. Note that this example shows VM file persistency, which does not work on clusters.
+
+=== Example: HTTP to VM
+
+[source,xml,linenums]
+----
+<!-- This is the reliable acquisition flow in the reliability pattern.  -->
+<flow name="reliable-data-acquisition">
+	<http:listener config-ref="HTTP_Listener_config" path="transactionalEndpoint"/>
+	<vm:publish config-ref="VM_Config" queueName="toTransactionalVM"/> <!--1-->
+</flow>
+
+<!-- This is the application logic flow in the reliability pattern.
+	It is a wrapper around a sub-flow, "business-logic-processing". -->
+<flow name="main-flow">
+	<vm:listener doc:name="Listener" config-ref="VM_Config" queueName="toTransactionalVM"
+		transactionalAction="ALWAYS_BEGIN"/> <!--2-->
+	<flow-ref name="business-logic-processing"/>
+</flow>
+
+<!-- This is where the real work of the application will happen. -->
+<sub-flow name="business-logic-processing">
+	<!--
+		This flow is where the actual business-logic is performed.
+	-->
+</sub-flow>
+----
+<1> The message is written to the VM queue. It is now available for processing by the main flow.
+<2> The message is read from the VM queue transactionally. This ensures that if an error occurs, the reading will be rolled back and the message reprocessed.
+
+
+== Implementing a Reliable Acquisition Flow
+
+The example provided in Implementing a Reliability Pattern showed a reliable acquisition flow for an HTTP listener to a VM publish-operation. Now, let's focus on a reliable acquisition flow that has a non-transactional Message Source: FTP-to-VM (Or SFTP, FTPS).
```

Comment:
Either make Implementing a Reliability Pattern a link, or use quotes to indicate it's the title. 
No hyphen between publish and operation.
I'm not sure about using the "let's" and "we"--it's not consistent with our style. Should be reworded so it's like "This scenario focuses on..." and "In this scenario a resource-based connection is used." etc etc